ISISScientificComputing/autoreduce
# Re-Run Jobs Validation

**Description of changes**

- Refactors converting string representations of a range of numbers into a separate file
- Adds new validation and better error messages when we do parse user output
- Incorrect input now returns an error to the user
- Run description and number of runs are limited by the server
- Travis CI Tests have been enabled for these changes
- Adds new run data to the populate DB SQL, so we can test multiple runs

- Minor: Removed unused imports where noticed


**How to test**
- Click re-run job:
- Enter a nonsense input such as 100-11a0
- For MUSR try any range of runs between 1 and 10, check it submits correctly
- Try an input with more than 500 runs (for admins), or less than 50 (for staff)
- Comment out the following lines in `run_variables.js` to disable client side checking:
```
            var max_length = 200;
            if($(this).val().length >= max_length) {
                addInvalid($(this), 'Re-run description must be less than ' + max_length.toString() + ' characters.')
            }
```
- Then try to enter a description more than 200 characters

Fixes #3 and #4 

---


**To test**
* Log into the devolpement nodes
* `git checkout this-pull-request-branch`
* restart services on each node (information on how to do this can be found in the development documentation)
* Submit some runs with the manualsubmission.py script

Once you are happy, merge this pull request into develop, and update the development nodes to the devlop branch
```
> git checkout origin/devlop
> git fetch -p
> git pull origin/develop
```
